nextjs-13/src/app/newsletter/[id]/page.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { ClipLoader } from 'react-spinners';
+
+import { useParams, redirect } from 'next/navigation';
+
+import NewsletterDetail from '@components/NewsletterDetail';
+import requestManager from 'lib/request/manager';
+
+const ViewNewsletter = () => {
+  const [record, setRecord] = useState([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const params = useParams();
+  const id = params.id;
+
+  const getNewletter = async () => {
+    setIsLoading(true);
+
+    try {
+      const response = await requestManager('GET', `v1/newsletter/${id}`);
+
+      setIsLoading(false);
+      setRecord(response.record);
+    } catch (error) {
+      setHasError(true);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getNewletter();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (hasError) redirect('/auth/sign-in');
```

Comment:
At first, I tried to use the `redirect` in [the catch block at line 29](https://github.com/nimblehq/js-framework-benchmark/pull/127/files#diff-aade95d258f37cabba282bbed7c140c258b15f86708d6234c8710fd15307bc0bR28), but Next.js raises [this error](https://github.com/vercel/next.js/issues/42556).

Seems like `redirect` can not be used in an async function(`fetchRecord`)?

So I added the state `hasError` and use `redirect` on this `useEffect` instead.